### PR TITLE
Check whether all nodes are created after OCP deployment

### PR DESCRIPTION
**Pull request by vavuthu**
Fixes: #4035 

Signed-off-by: vavuthu <vavuthu@redhat.com>

### CODE REVIEW

**Comment by vavuthu** [comment]
> PR validation
> 
> Cluster Name:
> Cluster Configuration: conf/ocsci/skip_ocs_deploy.yaml
> PR Test Suite: deployment
> PR Test Path: tests/
> Additional Test Params:
> OCP VERSION: 4.7
> OCS VERSION: 4.7
> 
> [Job UNSTABLE](https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-trigger-test-pr/402//testReport) (some or all tests failed).

Deployment o cluster is successfull

```
08:06:57 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-monitoring get Route prometheus-k8s -n openshift-monitoring -o yaml
08:06:57 - MainThread - /home/jenkins/workspace/qe-deploy-ocs-cluster/ocs-ci/ocs_ci/utility/prometheus.py - INFO - Generated CA certification file: /tmp/tmpf4h1osq6
-------------------------------- live log call ---------------------------------
08:06:57 - MainThread - tests.ecosystem.deployment.test_deployment - INFO - Verifying OCP cluster is running
08:06:57 - MainThread - ocs_ci.ocs.openshift_ops - INFO - Testing access to cluster with /home/jenkins/current-cluster-dir/openshift-cluster-dir/auth/kubeconfig
08:06:57 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc cluster-info
08:06:57 - MainThread - ocs_ci.ocs.openshift_ops - INFO - Access to cluster is OK!
PASSED                                                                   [100%]



-- Docs: https://docs.pytest.org/en/latest/warnings.html
- generated xml file: /home/jenkins/current-cluster-dir/logs/deployment_results_1616656520.xml -
- generated html file: file:///home/jenkins/current-cluster-dir/logs/deployment_report_1616656520.html -
========== 1 passed, 733 deselected, 2 warnings in 2680.50s (0:44:40) ==========

```

Job is failed during live run of deployment ( without --deploy option ) due to issue https://github.com/red-hat-storage/ocs-ci/issues/4050

Hence marking this PR as verified

File: ocs_ci/ocs/utils.py [hunk 1]
```diff
@@ -1132,3 +1136,26 @@ def reboot_node(ceph_node, timeout=300):
...
+    expected_num_nodes = (
+        ocsci_config.ENV_DATA["worker_replicas"]
+        + ocsci_config.ENV_DATA["master_replicas"]
+    )
+    existing_num_nodes = len(get_all_nodes())
+    if expected_num_nodes != existing_num_nodes:
+        log.error(
+            f"Expected number of nodes is {expected_num_nodes} but created during deployment is {existing_num_nodes}"
+        )
+        raise NotAllNodesCreated
```
Comment (petr-balogh) [inline ocs_ci/ocs/utils.py:1161]:
```suggestion
        raise NotAllNodesCreated(
            f"Expected number of nodes is {expected_num_nodes} but created during deployment is"
            f"{existing_num_nodes}"
            )
```

Comment (vavuthu) [reply ocs_ci/ocs/utils.py]: Done

File: ocs_ci/ocs/utils.py [hunk 2]
```diff
@@ -1132,3 +1136,25 @@ def reboot_node(ceph_node, timeout=300):
...
+def verify_all_nodes_created():
+    """
+    Verify all nodes are created or not
+
+    Raises:
+        NotAllNodesCreated: In case all nodes are not created
+
+    """
+    # importing here to avoid circular import
+    from ocs_ci.ocs.node import get_all_nodes
```
Comment (petr-balogh) [inline ocs_ci/ocs/utils.py:1150]: What about to move this function to ocs_ci.ocs.node so you will not have the issues with cyclical import and in the place where you would like to use it you will import from ocs_ci.ocs.node directly.

Comment (vavuthu) [reply ocs_ci/ocs/utils.py]: Done

**Review by petr-balogh — DISMISSED** [review]

**Review by shylesh — APPROVED** [review]
LGTM

**Review by petr-balogh — APPROVED** [review]

**Review by Akarsha-rai — APPROVED** [review]

**Pull request merged**